app/src/main/java/com/ataulm/whatsnext/di/AppComponent.kt

```diff
@@ -35,33 +33,28 @@ interface AppComponent {
     }
 }
 
-@Module(includes = [AppModule.Declarations::class])
+@Module
 internal object AppModule {
 
     @JvmStatic
     @Provides
-    fun whatsNextService(letterboxdApi: LetterboxdApi): WhatsNextService {
+    fun whatsNextService(application: Application): WhatsNextService {
+        val tokensStore = tokensStore(application)
+        val letterboxdApi = letterboxdApi(tokensStore)
         val filmSummaryConverter = FilmSummaryConverter()
         val filmRelationshipConverter = FilmRelationshipConverter()
-        return WhatsNextService(letterboxdApi, filmSummaryConverter, filmRelationshipConverter)
+        return WhatsNextService(letterboxdApi, tokensStore, filmSummaryConverter, filmRelationshipConverter)
     }
 
-    @JvmStatic
-    @Provides
-    fun letterboxdApi(application: Application): LetterboxdApi {
+   private fun letterboxdApi(tokensStore: TokensStore): LetterboxdApi {
         return LetterboxdApiFactory(
                 apiKey = BuildConfig.LETTERBOXD_KEY,
                 apiSecret = BuildConfig.LETTERBOXD_SECRET,
-                tokensStore = TokensStore.create(application),
+                tokensStore = tokensStore,
                 clock = Clock(),
                 enableHttpLogging = BuildConfig.DEBUG
         ).createRemote()
     }
 
-    @Module
-    interface Declarations {
-
-        @Binds
-        fun application(application: Application): Application
```

Comment:
I think this was broken on the [last PR where I changed it to something that didn't make sense](https://github.com/ataulm/whats-next/pull/10/files#diff-e4bc0555f9b2fcdbc90f939294b2b2a5L68-R65).

This used to say:

```kotlin
@Binds
fun context(application: Application): Context
``` 

so that whenever `Context` was requested, Dagger would look for `Application` and return that. We could have written it as:

```kotlin
@JvmStatic
@Provides
fun context(application: Application): Context = application
```

but since `Application` is assignable to type `Context`, Dagger doesn't need a method body (and this is the purpose of `@Binds`):

>whenever there is a @Provides whose implementation is simple and common enough to be inferred by Dagger, it makes sense to just declare that as a method without a body (an abstract method) and have Dagger apply the behavior.
// [dagger.dev](https://dagger.dev/dev-guide/faq.html#why-is-binds-different-from-provides)

---

In the last PR, I was like "well, if we know we want the application context as a dependency, then it makes sense to request `Application`, not `Context`" and then I just changed it to what was written here. But then it gets confused - how can it provide `Application` (the return type) when `Application` is a dependency?

actually, we don't need this. Since we have [this `@BindsInstance` on the component builder](https://github.com/ataulm/whats-next/pull/12/files#diff-e4bc0555f9b2fcdbc90f939294b2b2a5R29-R30):

```kotlin
@BindsInstance
fun application(application: Application): Builder
```

which makes `Application` available to the graph anyway.